src/vos/vos_gc.c

```diff
@@ -1509,13 +1536,50 @@ gc_close_pool(struct vos_pool *pool)
 	return gc_close_bkt(&pool->vp_gc_info);
 }
 
+static int
+dlck_pd_ext_check(struct vos_pool_ext_df *pd_ext, umem_off_t off, struct dlck_print *dp)
+{
+	DLCK_PRINTF(dp, "Pool extension (off=%#x)... ", off);
+
+	if (pd_ext == NULL) {
+		DLCK_PRINT_OK(dp);
+		return DER_SUCCESS;
+	}
+
+	for (int i = 0; i < VOS_POOL_EXT_DF_PADDING_SIZE; ++i) {
+		if (pd_ext->ped_paddings[i] != 0) {
+			DLCK_PRINTF_ERR(dp, "non-zero padding[%d] (%#" PRIx64 ")\n", i,
+					pd_ext->ped_paddings[i]);
+			return -DER_NOTYPE;
```

Comment:
That may happen if we run DLCK against downgraded device. I am not sure whether we will support downgrade case or not in the future, but it is better to avoid returning failure for unknown preserved/padding fields.